### PR TITLE
Tungsten transmission

### LINKED ISSUE #150: Include transmission through tungsten grids
Higher energy photons in the 50 - 70 keV and above 90 keV have a > 1% probability of passing through the tungsten of the grids. This should be included in the grid factor applied to the response matrix for analysis of larger flares.

### PR DESCRIPTION
**Pull request by grazwegian**
Include effect of tungsten grid transparency at high energies when calculating response matrix. 
Resolves #150

### CODE REVIEW

File: stix/idl/processing/subcollimator/stx_subc_transmission.pro [hunk 1]
```diff
@@ -6,65 +6,91 @@
...
+  restore,loc_file( 'grid_temp.sav', path = getenv('STX_GRID') )
+  fff=read_ascii(loc_file( 'grid_param_front.txt', path = getenv('STX_GRID') ),temp=grid_temp)
+  rrr=read_ascii(loc_file( 'grid_param_rear.txt', path = getenv('STX_GRID') ),temp=grid_temp)
+
+  transm = fltarr(n_elements(ph_in),32)
+
+  mass_attenuation = xsec(ph_in, 74,'AB',/cm2perg, error=error,use_xcom=1)
+  gmcm = 19.30
+
+  linear_attenuation = mass_attenuation*gmcm/10.
```
Comment (paolomassa) [inline stix/idl/processing/subcollimator/stx_subc_transmission.pro:52]: I would suggest adding a comment on what lines 49-52 represent

File: stix/idl/processing/subcollimator/stx_subc_transmission.pro [hunk 2]
```diff
@@ -6,65 +6,91 @@
...
+
+  theta = sqrt( ( flare_loc[0]* flare_loc[0]) + ( flare_loc[1]* flare_loc[1]) ) / 3600./!radeg
```
Comment (paolomassa) [inline stix/idl/processing/subcollimator/stx_subc_transmission.pro:54]: Can we rewrite this line as `theta = sqrt( flare_loc[0]^2. + flare_loc[1]^2. ) / 3600. * !dtor`? It's exactly the same, but  I think it's more clear

File: stix/idl/processing/subcollimator/stx_subc_transmission.pro [hunk 3]
```diff
@@ -6,65 +6,91 @@
...
+  for i=0,n_elements(grid_pitch_front)-1 do begin
+
+    ;; Exclude detectors 1 and 2
+    if (sc[i] ne 11) and (sc[i] ne 12) and (sc[i] ne 13) and (sc[i] ne 17) and (sc[i] ne 18) and (sc[i] ne 19) then begin
 
-restore,loc_file( 'grid_temp.sav', path = getenv('STX_GRID') )
-fff=read_ascii(loc_file( 'grid_param_front.txt', path = getenv('STX_GRID') ),temp=grid_temp)
-rrr=read_ascii(loc_file( 'grid_param_rear.txt', path = getenv('STX_GRID') ),temp=grid_temp)
+      transm_front = stx_grid_transmission(flare_loc[0], flare_loc[1], grid_orient_front[i], $
+        grid_pitch_front[i], grid_slit_front[i], grid_thick_front[i], bridge_width_front[i], bridge_pitch_front[i], linear_attenuation, flux = 1.)
```
Comment (paolomassa) [inline stix/idl/processing/subcollimator/stx_subc_transmission.pro:81]: I would remove `flux = 1.` (see also line 84) and add `default, flux, 1` at the beginning of `stx_grid_transmission`

File: stix/idl/processing/subcollimator/stx_subc_transmission.pro (hunk 2 above)
Comment (grazwegian) [reply stix/idl/processing/subcollimator/stx_subc_transmission.pro:54]: That's more clear but in checking I think we don't ever use theta here so I've removed it altogether.

File: stix/idl/processing/subcollimator/stx_subc_transmission.pro (hunk 3 above)
Comment (grazwegian) [reply stix/idl/processing/subcollimator/stx_subc_transmission.pro:81]: Good point this indeed shouldn't be fixed here.

File: stix/idl/processing/subcollimator/stx_subc_transmission.pro (hunk 1 above)
Comment (grazwegian) [reply stix/idl/processing/subcollimator/stx_subc_transmission.pro:52]: Good suggestion, I have added some description and moved the energy transmission stuff together.

File: stix/idl/processing/subcollimator/stx_grid_transmission.pro [hunk 4]
```diff
@@ -6,46 +6,84 @@
...
+  shadow_width_e = replicate(shadow_width, nenergies)
+  grid_pitch_e =  replicate(grid_pitch, nenergies)
+
+  effective_slit_width = grid_slit_e + shadow_width_e * (1. - 2.* (1- edge_transmission)/(1. - slat_transmission))
+
+  flux_calibration = 2*(slat_transmission + (1.- slat_transmission)*effective_slit_width*bridge_factor/grid_pitch_e)
+
+  amplitude_calibration = (1. - slat_transmission) * bridge_factor * sin(!pi * effective_slit_width / grid_pitch_e )
+
+  transmission = keyword_set(flux) ? flux_calibration : amplitude_calibration
```
Comment (samaloney) [inline stix/idl/processing/subcollimator/stx_grid_transmission.pro:84]: Is this documented somewhere, kinda see what's going on but not 100% sure.

File: stix/idl/processing/subcollimator/stx_grid_transmission.pro [hunk 5]
```diff
@@ -6,46 +6,84 @@
...
+  effective_slit_width = grid_slit_e + shadow_width_e * (1. - 2.* (1- edge_transmission)/(1. - slat_transmission))
+
+  flux_calibration = 2*(slat_transmission + (1.- slat_transmission)*effective_slit_width*bridge_factor/grid_pitch_e)
+
+  amplitude_calibration = (1. - slat_transmission) * bridge_factor * sin(!pi * effective_slit_width / grid_pitch_e )
+
+  transmission = keyword_set(flux) ? flux_calibration : amplitude_calibration 
+
+  ; factors are relative to 0.5 value for ideal grids
+  return, transmission/2.
```
Comment (samaloney) [inline stix/idl/processing/subcollimator/stx_grid_transmission.pro:87]: Was this also the case for the previous values?

**Review by samaloney — APPROVED** [review]

File: stix/idl/processing/subcollimator/stx_grid_transmission.pro (hunk 4 above)
Comment (grazwegian) [reply stix/idl/processing/subcollimator/stx_grid_transmission.pro:84]: The documentation is a writeup for Gordon, the Recipe for STIX Flux and Amplitude Calibration, it’s still in progress so not publicly available yet but I can send it to you directly so you can get more of a feeling for the calculation,

File: stix/idl/processing/subcollimator/stx_grid_transmission.pro (hunk 5 above)
Comment (grazwegian) [reply stix/idl/processing/subcollimator/stx_grid_transmission.pro:87]: Yes, the previous versions used the slit to pitch ratio which is 0.5 for ideal grids to approximate the grid transmission. This is also usually the dominant term for this calculation.

**Pull request merged**